src/test/rgw/test_d4n_directory.cc

```diff
@@ -200,6 +200,7 @@ TEST_F(ObjectDirectoryFixture, GetYield)
   io.run();
 }
 
+/* Does not currently pass on Ubuntu due to incompatible Redis version.
```

Comment:
commenting out the test cases won't help if users want to enable/run d4n on ubuntu. `D4NFilterObject::copy_object()` will just fail. how can we avoid sending COPY requests altogether?

Reply:
I can comment out the method entirely so it cannot be called.

Reply:
looking at copy_object(), it should probably not try to update the directory until the object copy operation has actually succeeded. at that point, rgw should have all the info it needs to call get_obj_dir()->set() instead of trying to copy

Reply:
p.s. until we have writeback caching, this should probably not update the directory at all. next->copy_object() will do the copy normally, and the next GetObject request on that copy will initialize the cache/directory

Reply:
Pritha has implemented an updated version of the `copy_object` logic for the single node cache, which is the next PR to be merged. Should I still go through with updating the logic for this method rather than removing the directory's `copy` call?

Reply:
I just saw your newer response. I will remove the call.